tensorboard/plugins/example/smoke_test.sh

```diff
@@ -44,18 +44,51 @@ pip install -qU wheel 'setuptools>=36.2.0'
 (cd ./example-plugin && python setup.py bdist_wheel)
 [ -f ./example-plugin/dist/*.whl ]  # just one wheel
 
-# This plugin doesn't require TensorFlow, so we don't install it.
 py_major_version="$(python -c 'import sys; print(sys.version_info[0])')"
+pip install tf-nightly-2.0-preview  # TODO(@wchargin): Other versions, too?
+pip uninstall -y tensorboard tb-nightly  # drop conflicting packages
 pip install ./tensorboard-wheels/*py"${py_major_version}"*.whl
 pip install ./example-plugin/dist/*.whl
 
+python -m tensorboard_plugin_example.demo
+
 # Test tensorboard + tensorboard_plugin_example integration.
 mkfifo pipe
-tensorboard --port=0 --logdir=smokedir 2>pipe &
+tensorboard --port=0 --logdir=. 2>pipe &
 perl -ne 'print STDERR;/http:.*:(\d+)/ and print $1.v10 and exit 0' <pipe >port
-curl -fs "http://localhost:$(cat port)/data/plugins_listing" >plugins_listing
-grep -F '"example":' plugins_listing
-grep -F '"/data/plugin/example/index.js"' plugins_listing
-curl -fs "http://localhost:$(cat port)/data/plugin/example/index.js" >index.js
+port="$(cat port)"
+curl -fs "http://localhost:${port}/data/plugin/example/index.js" >index.js
 diff -u example-plugin/tensorboard_plugin_example/static/index.js index.js
+
+# Wait for data to be read so that the plugin becomes active.
+for attempt in {1..5}; do
+  sleep 1
+  curl -fs "http://localhost:${port}/data/plugins_listing" >plugins_listing
+  status="$(python3 -c '
+import sys
+import json
+with open("plugins_listing") as infile:
+  plugins = json.load(infile)
+if "example" in plugins:
+  print("active" if plugins["example"]["enabled"] else "inactive")
+else:
+  print("missing")
+  sys.stderr.write("%s\n" % (plugins,))
+  ')"
```

Comment:
For sure. What’s awkward is that

```shell
  status="$(python3 -c '
    import sys
    ...
  ')"
```

would fail with an `IndentationError`.\* I personally prefer the current
indentation to the −2-space variant, because the opening and closing
single quote appear at the same level of indentation. For comparison, if
the code were

```shell
  status="$(
    some long command that needs \
        multiple lines
  )"
```

then I think the indentation would be unobjectionable. If you have
strong feelings, though, I’ll happily make this flush-left.

(Another alternative is something like

```shell
  status="$(<<EOF sed 's/^    //' | python3 -
    import sys
    ...
EOF
  )"
```

but (a) that still requires the last line’s `EOF` to be at column 0, and
(b) it’s brittle; moving this into a (sh) function would require
changing the `sed` command. It’s also far less obvious.)

(And, yes, `<<"    EOF" ...` does work, but… ugh.)

\* IMHO, this is one major point against significant whitespace: yes,
code written by and for humans should be properly indented, but
(similarly) it can be easier for machines to [generate poorly formatted
code and then autoformat it][1], and you can’t do that if the poor
formatting is an actual syntax error. (To be fair, Haskell has
significant whitespace and will simply strip any uniform leading
whitespace from all lines, so the problem isn’t fundamental.)

[1]: https://github.com/sourcecred/sourcecred/blob/f934735afc97a6bb5015bec3b3d23f85818c6531/src/graphql/generateFlowTypes.js#L127-L132